client/hwlib/src/collectors/os_info.rs

```diff
@@ -42,12 +41,19 @@ impl CommandRunner for SystemCommandRunner {
 
 impl OS {
     pub(crate) fn try_new(
+        os_release_filepath: &Path,
         proc_version_filepath: &Path,
         runner: &impl CommandRunner,
     ) -> Result<Self> {
-        let codename = get_codename(runner)?;
-        let distributor = get_distributor(runner)?;
-        let version = get_version(runner)?;
+        let release = OsRelease::new_from(os_release_filepath).with_context(|| {
+            format!("cannot read OS release information from: {os_release_filepath:?}",)
+        })?;
+        let OsRelease {
+            version_codename: codename,
+            name: distributor,
+            version_id: version,
+            ..
+        } = release;
```

Comment:
I think the two new `let` statements form a block of strongly-associated code, hence a newline just here could be beneficial